src/main/scala/za/co/absa/spark/commons/implicits/StructTypeImplicits.scala

```diff
@@ -0,0 +1,360 @@
+      helper(pathToks, Seq()).mkString(".")
+    }
+
+    /**
+     * Get all array columns' paths out of complete path.
+     *
+     * E.g. if the path argument is "a.b.c.d.e" where b and d are arrays, "a.b" and "a.b.c.d" will be returned.
+     *
+     * @param path The path to the attribute
+     * @return Seq of dot-separated paths for all array fields in the provided path
+     */
+    def getAllArraysInPath(path: String): Seq[String] = {
+      @tailrec
+      def helper(remPath: Seq[String], pathAcc: Seq[String], arrayAcc: Seq[String]): Seq[String] = {
+        if (remPath.isEmpty) arrayAcc else {
+          val currPath = (pathAcc :+ remPath.head).mkString(".")
+          val currType = getFieldType(currPath)
+          currType match {
+            case Some(_: ArrayType) =>
+              val strings = pathAcc :+ remPath.head
+              helper(remPath.tail, strings, arrayAcc :+ strings.mkString("."))
+            case Some(_) => helper(remPath.tail, pathAcc :+ remPath.head, arrayAcc)
+            case None => arrayAcc
+          }
+        }
+      }
+
+      val pathToks = path.split("\\.")
+      helper(pathToks, Seq(), Seq())
+    }
+
+    /**
+     * For a given list of field paths determines the deepest common array path.
+     *
+     * For instance, if given 'a.b', 'a.b.c', 'a.b.c.d' where b and c are arrays the common deepest array
+     * path is 'a.b.c'.
+     *
+     * If any of the arrays are on diverging paths this function returns None.
+     *
+     * The purpose of the function is to determine the order of explosions to be made before the dataframe can be
+     * joined on a field inside an array.
+     *
+     * @param fieldPaths A list of paths to analyze
+     * @return Returns a common array path if there is one and None if any of the arrays are on diverging paths
+     */
+    def getDeepestCommonArrayPath(fieldPaths: Seq[String]): Option[String] = {
+      val arrayPaths = fieldPaths.flatMap(path => getAllArraysInPath(path)).distinct
+
+      if (arrayPaths.nonEmpty && isCommonSubPath(arrayPaths: _*)) {
+        Some(arrayPaths.maxBy(_.length))
+      } else {
+        None
+      }
+    }
+
+    /**
+     * For a field path determines the deepest array path.
+     *
+     * For instance, if given 'a.b.c.d' where b and c are arrays the deepest array is 'a.b.c'.
+     *
+     * @param fieldPath A path to analyze
+     * @return Returns a common array path if there is one and None if any of the arrays are on diverging paths
+     */
+    def getDeepestArrayPath(fieldPath: String): Option[String] = {
+      val arrayPaths = getAllArraysInPath(fieldPath)
+
+      if (arrayPaths.nonEmpty) {
+        Some(arrayPaths.maxBy(_.length))
+      } else {
+        None
+      }
+    }
+
+    /**
+     * Get paths for all array fields in the schema
+     *
+     * @return Seq of dot separated paths of fields in the schema, which are of type Array
+     */
+    def getAllArrayPaths(): Seq[String] = {
```

Comment:
If we keep that array functions in `SparkCommons` (for example because they are used both in Standardization and Conformance), I would probably put them to its own `implicit class`.

Reply:
Moving